From 18387447e11ee584d1a7d25ca11d5402f5c579da Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 18 Jan 2016 17:54:57 +0800
Subject: [PATCH 0856/5242] MLK-12165: ASoC: fsl: imx-cs42888: remove 32k and
 64k support

commit  02b889b1ea1f9e88d14103b8c356451da12d2fd5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Revert "MLK-11623 ASoC: imx-cs42888: add 32k and 64k sample rate support"

This reverts commit 314a01f40599134086480ef3c5e89a54aeedbf1f.

In Async mode, record and playback use different samplerate, one is 32k,
another is 48kHz, there will be issue "unsupported sysclk ratio".

example case is
arecord -Dhw:0,1 -f S16_LE -r 32000 -c 2 | aplay -f S16_LE -r 32000 -c 2

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-cs42888.c |   12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/sound/soc/fsl/imx-cs42888.c b/sound/soc/fsl/imx-cs42888.c
index dea5a9a..82d56d5 100644
--- a/sound/soc/fsl/imx-cs42888.c
+++ b/sound/soc/fsl/imx-cs42888.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2010-2015 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2010-2016 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -100,13 +100,11 @@ static int imx_cs42888_surround_startup(struct snd_pcm_substream *substream)
 	int ret;
 
 	if (priv->mclk_freq == 24576000) {
-		support_rates[0] = 32000;
-		support_rates[1] = 48000;
-		support_rates[2] = 64000;
-		support_rates[3] = 96000;
-		support_rates[4] = 192000;
+		support_rates[0] = 48000;
+		support_rates[1] = 96000;
+		support_rates[2] = 192000;
 		constraint_rates.list = support_rates;
-		constraint_rates.count = 5;
+		constraint_rates.count = 3;
 
 		ret = snd_pcm_hw_constraint_list(runtime, 0, SNDRV_PCM_HW_PARAM_RATE,
 							&constraint_rates);
-- 
1.7.9.5

